### ISSUE TITLE
LSP

### ISSUE THREAD

**Issue by Rudxain** [FusionSid/Rickroll-Lang-VScode-Extension#3]
I'll be honest, I have no idea how to write an [LSP](https://en.wikipedia.org/wiki/Language_Server_Protocol) 💀. But this would benefit RR, for the same reason as LLVM.

The extension would act as a "bridge" (interface) between VSC and RR-LSP, which is how `rust-analyzer` and Golang-ext work (please, anyone, correct me if I'm wrong). So the extension would still exist, but now we'll have an "universal" API that other code-editors (such as [`hx`](https://github.com/helix-editor/helix), and [NVim](https://github.com/neovim/neovim)) could use

**Comment by FusionSid** [comment]
> I have no idea how to write an [LSP](https://en.wikipedia.org/wiki/Language_Server_Protocol) 💀

same here.
When making this extension, I tried reading the [guide](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide) but didn't really understand it 😭.
I definitely agree that it would benefit rickroll-lang, just I have no idea how to make one.
I do have school holidays coming up in a week, so I guess I could look into it again.

**Comment by Rudxain** [comment]
> school holidays coming up in a week, so I guess I could look into it again.

Thank you in advance! I'll also try to learn LSP, but I'll write drafts when I can get my hands on my dev PC

**Comment by Rudxain** [comment]
Wait, I just realized, we'll have to delay the LSP, because of Rick-Lang/rickroll-lang#76, as the Rust implementation might end up being a [_slightly different_ language](https://github.com/Rick-Lang/rickroust/issues/4)

**Comment by SatinWukerORIG** [comment]
You are right. Also I am very sorry that I don't have time to maintain this language regularly.

**Comment by FusionSid** [comment]
omg i completely forgot about the LSP :sob:

**Comment by Rudxain** [comment]
> I am very sorry that I don't have time to maintain this language regularly

Don't worry, I've been (and still am) busy :sweat_smile:

> omg i completely forgot about the LSP 😭

Me too, lol
